### DISCUSSION TITLE
Ferramentas para teste de integração

### DISCUSSION THREAD

**Discussion by felipeorlando** [frontendbr/forum#547] [Perguntas e Respostas]
Galera, gostaria de saber quais ferramentas vocês usam e recomendam para fazer testes de integração no front. 

É importante que essa ferramenta não dependa de nenhuma outra e seja agnóstica a frameworks/libs. 😄

**Comment by abalad** [comment]
@felipeorlando O Que Seria 

> Agnóstica a frameworks/libs

"Teísmo agnóstico é o ponto de vista filosófico, que engloba ambos o teísmo e o agnosticismo. Um agnóstico teísta acredita na existência de pelo menos uma divindade, mas diz respeito à base desta proposição como "algo desconhecido ou inerentemente incognoscível."

[Definição Wikipedia](https://pt.wikipedia.org/wiki/Te%C3%ADsmo_agn%C3%B3stico)

**Comment by felipeorlando** [comment]
@abalad
https://snook.ca/archives/javascript/be_library_agnostic

Não depende de frameworks ou libs, mas pode ser usando para com elas. 👍

**Comment by klarkc** [comment]
Eu uso muito o NightwatchJS, mas tem outras como o PhantomJS, o próprio Mocha pode rodar no Browser, com algum trabalho para isso.

**Comment by guilhermejcgois** [comment]
Adicionando ao que o @klarkc falou, dá para usar o [NightwatchJS](http://nightwatchjs.org/) com o [PhantomJS](http://phantomjs.org/) também, porém o PhantomJS até onde eu sei é feito em cima do webkit, então pode ter problemas com algumas features mais novas, [Intl](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Intl) por exemplo (não consegui abrir o [lock do Auth0](https://auth0.com/docs/libraries/lock/v10) por conta disto). Outra coisa é que se você quiser testar com o firefox tenha em mente que o [webdriver da mozilla](https://developer.mozilla.org/en-US/docs/Mozilla/QA/Marionette/WebDriver) [não possui implementado](https://developer.mozilla.org/en-US/docs/Mozilla/QA/Marionette/WebDriver/status) todas as features da [especificação](https://w3c.github.io/webdriver/webdriver-spec.html), o mais completo até onde eu sei é o do [chrome](https://sites.google.com/a/chromium.org/chromedriver/getting-started). Além de que o Nightwatch você consegue rodar usando [mocha](https://github.com/nightwatchjs/nightwatch-docs/blob/master/guide/running-tests/using-mocha.md), mas nunca tentei.
